apps/service/src/utils/getRankingRounds.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ *
+ * 내 순위표에 나올 n강의 종류에 대한 배열을 반환합니다.
+ * eg) 16강 월드컵을 종료한 유저의 경우, 8강 4강이 나타나야 하므로 [0, 8, 4]를 반환 (0은 전체)
+ *
+ * @param round 유저가 실제로 진행한 월드컵이 몇 강인지
+ * @returns 내 순위표에 나올 n강 종류 eg) [0, 8, 4]
+ *
+ */
+
+export const getRankingRounds = (round: number) => {
+  const maxIndex = Math.log2(round);
+  const maxIndexArray = [...Array(maxIndex)].map((v, i) => Math.pow(2, i));
```

Comment:
```suggestion
  const maxIndexArray = [...Array(maxIndex)].map((_, i) => 2 ** i);
```

중요한 부분은 아니지만,, 의견이 궁금해서 여쭈어봅니당

위 써제션과 아래 써제션중에 뭐가 더 깔끔해보이시나요?

그리고 Array 생성자에 new를 붙이시는 편이신가요?
Array() 는 암묵적으로 new 를 붙여주는걸로 알고 있는데, 명시해주는게 좋을까요?

전 아래 코드가 더 직관적이란 생각이 드는데.. 혹시 더 생각해보면 좋을 부분이 있을까요?

Reply:
성능상 유의미한 차이가 아니라, 개인적인 취향들로 갈릴 것 같은데

저는 Array from 으로 하는 편입니당! 그 이유는 한글로 해석하면서 읽을 때 개인적으로 편하더라고요!

Array.from => 새로운 배열을 무엇으로부터 만든다

[...Array()] 어떤 새로운 배열을 만들고 spread 한 뒤, 특정 작업을 한다

이렇게 해석을 하는데, 전자가 남의 코드를 읽을 때 해석이 잘 되는 것 같은 기분탓이 있어서 전자로 사용합니당

Reply:
Math.pow와 ** 연산자도 비슷한 이유로 전자로 사용하는데

** 을 잘 보면 좋겠지만, 피곤하면 * 로 해석할 수도 있을 것 같아서

좀 더 의미 전달 차원에서 Math.pow로 사용하는 것 같아요!

Reply:
new를 붙이는 경우와 안 붙이는 경우를 저는

new 키워드는 기존 변수로부터 만드는게 아니라, 새로 무언가를 만들 때 사용하고,

new를 쓰지 않는 경우는 기존 변수를 형변환을 이용하려 할 때 사용하는 것 같아요!

Reply:
@poiu694 호오 그렇게 생각할 수도 있겠네요! 자세한 의견 너무 감사합니다..!! 🥺 갬동